### PR TITLE
Fix/icon issues

### PR DESCRIPTION
**Pull request by bainternet**
- Added `_update_widget_settings` `_migrate_icon_fa4_value`
- added `fa4_to_fa5_value_migration` and migrate on import
- added `_v_2_6_6_fa4_migration_button`
- fixed on import icon migration

### CODE REVIEW

File: core/upgrade/upgrade-utils.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	 * @return mixed
+	 */
+	public static function _migrate_icon_fa4_value( $element, $args ) {
+		$widget_id = $args['widget_id'];
+
+		if ( empty( $element['widgetType'] ) || $widget_id !== $element['widgetType'] ) {
+			return $element;
+		}
+		foreach ( $args['control_ids'] as $old_name => $new_name ) {
+			if ( ! empty( $element['settings'][ $old_name ] ) ) {
```
Comment (matipojo) [inline core/upgrade/upgrade-utils.php]: `!` !!!

File: core/upgrade/upgrade-utils.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+				continue;
+			}
+			$element['settings'][ $new_name ] = Icons_Manager::fa4_to_fa5_value_migration( $element['settings'][ $old_name ] );
```
Comment (matipojo) [inline core/upgrade/upgrade-utils.php]: if ( ! isset( $element['settings'][ $new_name ]  )

File: includes/managers/icons.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ public function register_styles() {
 				continue;
 			}
 			$dependencies = [];
-			if ( isset( $icon_type['enqueue'] ) ) {
+			if ( isset( $icon_type['enqueue'] ) && $icon_type['enqueue'] ) {
```
Comment (matipojo) [inline includes/managers/icons.php]: if ( ! empty( $icon_type['enqueue'] ) ) {

File: core/upgrade/upgrade-utils.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	/**
+	 * migrate Icon control string value to Icons control array value
+	 *
+	 * @param array $element
+	 * @param array $args
+	 *
+	 * @return mixed
+	 */
+	public static function _migrate_icon_fa4_value( $element, $args ) {
```
Comment (matipojo) [inline core/upgrade/upgrade-utils.php]: Move `_migrate_icon_fa4_value` to core/upgrade/upgrades.php

Comment (bainternet) [reply core/upgrade/upgrade-utils.php]: why?

File: core/upgrade/upgrade-utils.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Elementor\Core\Upgrade;
+
+use Elementor\Icons_Manager;
```
Comment (matipojo) [inline core/upgrade/upgrade-utils.php]: Remove

**Pull request merged**